components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/vault/VaultTool.java

```diff
@@ -195,4 +210,24 @@ private static String doEncryption(Cipher cipher, String plainTextPwd) {
         System.out.println("\nEncryption is done Successfully\n");
         return encodedValue;
     }
+
+    private static void addSystemProperties() {
+        Properties properties = new Properties();
+        String filePath = System.getProperty(CONF_DIRECTORY_PATH) + File.separator + CARBON_PROPERTIES;
+        File file = new File(filePath);
+
+        if (file.exists()) {
+            try ( InputStream inputStream = new FileInputStream(file)) {
+                properties.load(inputStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.exit(1);
+            }
+        }
+
+        Set<Object> keys = properties.keySet();
+        for (Object key: keys)  {
+            System.setProperty((String)key, (String)properties.get(key));
```

Comment:
Do we need to load all properties from carbon.properties? Isn't org.wso2.SecureVaultPasswordRegEx sufficient?

Reply:
Changed to read org.wso2.SecureVaultPasswordRegEx property